# Notebook 1: Data Ingestion & Overview
**Project**: Loan Approval Prediction & Banking Analytics
**Author**: Banking Risk Analytics Team

---
## 1. Introduction
This notebook handles the initial data ingestion and overview. We load the raw **Loan Prediction Dataset** and perform a high-level review of the columns, dimensions, data types, and initial summary statistics.

### Tech Stack:
- Python
- Pandas
- NumPy


In [1]:
import pandas as pd
import numpy as np
import os

print("Pandas version:", pd.__version__)
print("NumPy version:", np.__version__)

Pandas version: 2.3.3
NumPy version: 2.3.5


## 2. Ingesting the Raw Dataset

In [2]:
# Path to the raw CSV file
raw_data_path = os.path.join("..", "Dataset", "raw", "loan_prediction.csv")

# Ingest dataset
df = pd.read_csv(raw_data_path)
print("Dataset loaded successfully!")

Dataset loaded successfully!


## 3. Dataset Dimensions and Overview

In [3]:
# Check shape of the dataset
print(f"Number of rows (applicants): {df.shape[0]}")
print(f"Number of columns (features): {df.shape[1]}")

Number of rows (applicants): 614
Number of columns (features): 13


In [4]:
# View first 5 rows
df.head()

,Loan_ID,Gender,Married,Dependents,Education,Self_Employed,ApplicantIncome,CoapplicantIncome,LoanAmount,Loan_Amount_Term,Credit_History,Property_Area,Loan_Status
0,LP001002,Male,No,0,Graduate,No,5849,0.0,NaN,360.0,1.0,Urban,Y
1,LP001003,Male,Yes,1,Graduate,No,4583,1508.0,128.0,360.0,1.0,Rural,N
2,LP001005,Male,Yes,0,Graduate,Yes,3000,0.0,66.0,360.0,1.0,Urban,Y
3,LP001006,Male,Yes,0,Not Graduate,No,2583,2358.0,120.0,360.0,1.0,Urban,Y
4,LP001008,Male,No,0,Graduate,No,6000,0.0,141.0,360.0,1.0,Urban,Y


## 4. Metadata and Data Types

In [5]:
# View column data types and null counts
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


## 5. Descriptive Statistics

### Summary Statistics for Numerical Columns

In [6]:
df.describe()

,ApplicantIncome,CoapplicantIncome,LoanAmount,Loan_Amount_Term,Credit_History
count,614.000000,614.000000,592.000000,600.00000,564.000000
mean,5403.459283,1621.245798,146.412162,342.00000,0.842199
std,6109.041673,2926.248369,85.587325,65.12041,0.364878
min,150.000000,0.000000,9.000000,12.00000,0.000000
25%,2877.500000,0.000000,100.000000,360.00000,1.000000
50%,3812.500000,1188.500000,128.000000,360.00000,1.000000
75%,5795.000000,2297.250000,168.000000,360.00000,1.000000
max,81000.000000,41667.000000,700.000000,480.00000,1.000000


### Summary Statistics for Categorical Columns

In [7]:
df.describe(include=['object'])

,Loan_ID,Gender,Married,Dependents,Education,Self_Employed,Property_Area,Loan_Status
count,614,601,611,599,614,582,614,614
unique,614,2,2,4,2,2,3,2
top,LP001002,Male,Yes,0,Graduate,No,Semiurban,Y
freq,1,489,398,345,480,500,233,422


## 6. Key Data Quality Observations
Based on the high-level description:
1. **Missing Values**: Features like `Gender`, `Married`, `Dependents`, `Self_Employed`, `LoanAmount`, `Loan_Amount_Term`, and `Credit_History` contain missing values (nulls) that need to be resolved.
2. **Data Types**: `Credit_History` is loaded as float64, but represents a discrete binary category (0 or 1).
3. **Outliers**: The max value for `ApplicantIncome` ($81,000) and `CoapplicantIncome` ($41,667) is significantly higher than the median, suggesting high right-skewness and outliers.
